src/audio/kpb.c

```diff
@@ -336,9 +340,9 @@ static int kpb_unbind(struct comp_dev *dev, void *data)
 
 	/* Reset sinks when unbinding */
 	if (buf_id == 0)
```

Comment:
Probably no need to check for `buf_id == 0 `, just assign NULL based on `src_queue`?